app/src/main/java/com/realityexpander/tasky/presentation/splash_screen/SplashScreen.kt

```diff
@@ -0,0 +1,148 @@
+import com.ramcosta.composedestinations.navigation.DestinationsNavigator
+import com.realityexpander.tasky.R
+import com.realityexpander.tasky.common.settings.AppSettings
+import com.realityexpander.tasky.common.settings.setSettingsLoaded
+import com.realityexpander.tasky.dataStore
+import com.realityexpander.tasky.domain.AuthInfo
+import com.realityexpander.tasky.presentation.common.modifiers.largeHeight
+import com.realityexpander.tasky.presentation.destinations.AgendaScreenDestination
+import com.realityexpander.tasky.presentation.destinations.LoginScreenDestination
+import com.realityexpander.tasky.presentation.ui.theme.TaskyTheme
+import com.realityexpander.tasky.presentation.common.util.UiText
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+
+@Composable
+@Destination
+@RootNavGraph(start = true)
+fun SplashScreen(
+    navigator: DestinationsNavigator,
+    viewModel: SplashViewModel = hiltViewModel()
+) {
+    val splashState by viewModel.splashState.collectAsState()
+    val scope = rememberCoroutineScope()
+    val context = LocalContext.current
+
+    val appSettings = context.dataStore.data.collectAsState( // .data is a Flow
+        initial = AppSettings()
+    ).value
+
+    // Load settings from data store
+    LaunchedEffect(key1 = appSettings) {
+        // ignore first data (initial emit is always default state, due to it being a flow)
+        if(!appSettings.settingsLoaded) {
+            context.dataStore.setSettingsLoaded(true)
+            return@LaunchedEffect
+        }
+
+        viewModel.onSetAuthInfo(
+            AuthInfo(
+                authToken = appSettings.authInfo.authToken,
+                userId = appSettings.authInfo.userId,
+                username = appSettings.authInfo.username,
+            )
+        )
+    }
+
+    // After settings are Loaded (or not) - Navigate to correct screen
+    if (splashState.authInfoChecked) {
+        splashState.authInfo?.authToken?.let { authToken ->
+            if (authToken != AuthInfo.NOT_LOGGED_IN.authToken) {
+                navigator.navigate(
+                    AgendaScreenDestination(
+//                         authInfo = splashState.authInfo // todo add authInfo to agenda screen
+                    )
+                )
+            }
+        } ?: run {
+            scope.launch {
+                delay(500) // show logo for .5 second before going to login screen
+
+                navigator.navigate(
+                    LoginScreenDestination(
+                        username = "Chris Athanas",     // TESTING ONLY
+                        email = "chris3@demo.com",      // TESTING ONLY
+                        password = "Password1",         // TESTING ONLY
+                        confirmPassword = "Password1",  // TESTING ONLY
+                    )
+                ) {
+                    launchSingleTop = true
+                    restoreState = true
+                }
+            }
+        }
+    }
```

Comment:
This whole thing is a side-effect as well, makes more sense to use a `LaunchedEffect` block here with `authToken` passed as key.

Reply:
I have removed this in favor of using the Android 12 splash screen.